operator/controllers/release_manager.go

```diff
@@ -0,0 +1,192 @@
+package controllers
+
+import (
+	shipitv1beta1 "ship-it-operator/api/v1beta1"
+
+	"k8s.io/client-go/tools/record"
+	"k8s.io/helm/pkg/helm"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+	"k8s.io/helm/pkg/proto/hapi/release"
+)
+
+// HelmReleaseFinalizer allows the controller to clean up the associated release
+// before the HelmRelease resource is deleted.
+const HelmReleaseFinalizer = "HelmReleaseFinalizer"
+
+// ReleaseManager manages the lifecycle of a release using a helm client and
+// modifies HelmRelease fields like the Finalizers and Status. It broadcasts
+// kubernetes events whenever the release's status condition changes.
+type ReleaseManager struct {
+	helm     HelmClient
+	recorder record.EventRecorder
+	release  shipitv1beta1.HelmRelease
+}
+
+func (m *ReleaseManager) updateCondition(cond shipitv1beta1.HelmReleaseCondition) *shipitv1beta1.HelmRelease {
+	m.release.Status.SetCondition(cond)
+	// "Normal" is the default kubernetes event type
+	m.recorder.Event(&m.release, "Normal", cond.Type, cond.Message)
+
+	return m.release.DeepCopy()
+}
+
+func (m *ReleaseManager) ChartSpec() *shipitv1beta1.ChartSpec {
```

Comment:
I chose to add these 3 getter methods rather than making `ReleaseManager.release` a public field, so that it's not possible to mutate the `ReleaseManager`'s internal release state through one of the returned pointers.